### ISSUE TITLE
Expose `EscapeSequenceParser` to public API

### ISSUE THREAD

**Issue by JounQin** [xtermjs/xterm.js#4191]
I want to parse output text from socket directly and get displayed text like xterm.

There is a similar issue previously #3594, but I also did not find the solution.

**Comment by jerch** [comment]
> ... and get displayed text like xterm

The most straight forward way is to use xterm.js for that (as it kinda guarantees, that most things will be handled correctly) - and pull the content from the terminal buffer exposed. Exposing the parser directly does not help you much, as it means that you have to implement tons of low level handling yourself (basically most of a TE on top of the sequence parser). If thats really what you want w'o using the other convenient functionality of xterm.js - it is OSS MIT licensed, thus you can pull the parser from the repo and use it to any extend you want.
So nope, we are not likely to expose the parser directly anytime soon, as it is not the right way to interact with xterm.js for >90% of the lib users.

**Comment by JounQin** [comment]
> Care to explain whats your real use case here?

We want to send `pwd` command and get the correct output and display it out of terminal.

**Comment by jerch** [comment]
> We want to send `pwd` command and get the correct output and display it out of terminal (what means we don't want to display it in xterm temporarily).

This is almost impossible to get done right - not because of the missing parser symbols from xterm.js (thats not really an issue with `pwd` as it only returns a straight printable string), but because of the way a PTY works - you never can be sure, that a shell understanding the `pwd` command is currently the foreground process, also the output might be more than what `pwd` returns from another command currently running in the background, even being interleaved if returned data length its greater than PIPE_BUF (4096 on most OS these days).
The interleaving return data can be somewhat worked around by using sentinels, e.g. getting the `pwd` output from a small shell script like `echo "PWDSTART$(pwd)PWDEND"` and searching/cutting at those sentinels, if the pwd output+sentinels itself stays below PIPE_BUF size. But it still needs the shell in the foreground, which cannot be ensured/tested for ssh connections by any means.

Long story short - any attempt to get this working reliable is doomed to fail. We have no exclusive process-to-pty binding in POSIX terminal interface.

**Comment by JounQin** [comment]
@jerch Thanks for your great explanation! In my case, we don't need to 100% correct output, we want to cover most(for example 80) user cases, do you think that's possible？

**Comment by jerch** [comment]
@JounQin Sure you can try that. I suggest to operate directly on the pty interface for that (needs just some regexp on data packages), this way xterm.js does not see any of it and you dont have to bother with deleting chars and cursor re-positioning.
